### PR TITLE
Obsolete the ReadUntypedAsString Resolve the build warning for StringComparison

### PR DESCRIPTION
**Pull request by xuzhg**
Obsolete the ReadUntypedAsString
Resolve the build warning for StringComparison

### CODE REVIEW

File: src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace Microsoft.OData.Client.Metadata
     using System.Reflection;
     using Microsoft.OData.Metadata;
     using Microsoft.OData.Edm;
-    using c = Microsoft.OData.Client;
+    using C = Microsoft.OData.Client;
```
Comment (mikepizzo) [inline src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs:20]: In other places you use "Client"

File: src/Microsoft.OData.Client/Metadata/ODataTypeInfo.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.OData.Client.Metadata
     using System.Diagnostics;
     using System.Linq;
     using System.Reflection;
-    using c = Microsoft.OData.Client;
+    using C = Microsoft.OData.Client;
```
Comment (mikepizzo) [inline src/Microsoft.OData.Client/Metadata/ODataTypeInfo.cs:18]: "Client"?

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs [hunk 3]
```diff
@@ -67,7 +67,7 @@ public ODataMessageReaderSettings(ODataVersion odataVersion)
         }
 
         /// <summary>
-        /// Gets or sets library compatibility version. Default value is <see cref="Microsoft.OData.ODataLibraryCompatibility.Latest"/>,
+        /// Gets or sets library compatibility version. Default value is <see cref="Microsoft.OData.ODataLibraryCompatibility.None"/>,
```
Comment (mikepizzo) [inline src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:70]: did we change this in 8.x? why?

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs [hunk 4]
```diff
@@ -194,6 +194,7 @@ public ODataMessageQuotas MessageQuotas
         /// <summary>
         /// Whether to read untyped values as a raw string.
         /// </summary>
+        [Obsolete("This will be dropped in the 9.x release. Please use 'PrimitiveTypeResolver' for unknown types.")]
```
Comment (mikepizzo) [inline src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:197]:
```suggestion
        [Obsolete("ReadUntypedAsString is deprecated and will be dropped in a future release. Please use 'PrimitiveTypeResolver' for primitive unknown types.")]
```

File: src/Microsoft.OData.Core/ODataMessageWriterSettings.cs [hunk 5]
```diff
@@ -114,7 +114,7 @@ public ODataMessageWriterSettings(ODataVersion? version)
         }
 
         /// <summary>
-        /// Gets or sets library compatibility version. Default value is <see cref="Microsoft.OData.ODataLibraryCompatibility.Latest"/>,
+        /// Gets or sets library compatibility version. Default value is <see cref="Microsoft.OData.ODataLibraryCompatibility.None"/>,
```
Comment (mikepizzo) [inline src/Microsoft.OData.Core/ODataMessageWriterSettings.cs:117]: 8.x change?

File: src/Microsoft.OData.Edm/ExtensionMethods/ToTraceStringExtensionMethods.cs [hunk 6]
```diff
@@ -136,6 +137,7 @@ private static void AppendFacets(this StringBuilder sb, IEdmPrimitiveTypeReferen
             }
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1305:Specify IFormatProvider", Justification = "<Pending>")]
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/ExtensionMethods/ToTraceStringExtensionMethods.cs:140]: should this have something here?

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/SemanticAst/ODataPathExtensionsTests.cs [hunk 7]
```diff
@@ -10,7 +10,7 @@
 using Microsoft.OData.Metadata;
 using Microsoft.OData.UriParser;
 using Xunit;
-using mbh = Microsoft.OData.Tests.UriParser.ModelBuildingHelpers;
+using Mbh = Microsoft.OData.Tests.UriParser.ModelBuildingHelpers;
```
Comment (mikepizzo) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/SemanticAst/ODataPathExtensionsTests.cs:13]: maybe call this "Helper"?

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 3 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:70]: There's no 'Latest' enum member for ODataLibraryCompatibility, And I noticed that in the constructor, 'None' is used for default value. I didn't see any changes within 8.x version for this part..

File: src/Microsoft.OData.Core/ODataMessageWriterSettings.cs (hunk 5 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ODataMessageWriterSettings.cs:117]: There's no 'Latest' enum member for ODataLibraryCompatibility, And I noticed that in the constructor, 'None' is used for default value. I didn't see any changes within 8.x version for this part..

File: src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs:20]: different idea in different day.  
😄 Let me change to use 'Client'.

File: src/Microsoft.OData.Client/Metadata/ClientPropertyAnnotation.cs [hunk 8]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.OData.Client.Metadata
     using System.Linq.Expressions;
     using System.Reflection;
     using Microsoft.OData.Edm;
-    using c = Microsoft.OData.Client;
+    using C = Microsoft.OData.Client;
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply src/Microsoft.OData.Client/Metadata/ClientPropertyAnnotation.cs:17]: changed to 'Client'

File: src/Microsoft.OData.Client/Metadata/ODataTypeInfo.cs (hunk 2 above)
Comment (xuzhg) [reply src/Microsoft.OData.Client/Metadata/ODataTypeInfo.cs:18]: #resolved

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 3 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:70]: I think that was changed in one of 7 version.

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 4 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:197]: Thanks. I changed using your sentence.

By the way, I am using the 'same' sentence' following up the existing 'Obsolete'. @challenh Can we consider changing others?

File: src/Microsoft.OData.Core/ODataMessageWriterSettings.cs (hunk 5 above)
Comment (mikepizzo) [reply src/Microsoft.OData.Core/ODataMessageWriterSettings.cs:117]: Okay; found Elizabeth's [commit ](https://github.com/OData/odata.net/commit/fde7d8e0f02879799ac06a4de3dddc2613d054af) that changed this. Thanks for catching the comment.

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 3 above)
Comment (mikepizzo) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:70]: found the [commit ](https://github.com/OData/odata.net/commit/fde7d8e0f02879799ac06a4de3dddc2613d054af) that updated this. Thanks for catching the comment!

File: src/Microsoft.OData.Edm/ExtensionMethods/ToTraceStringExtensionMethods.cs (hunk 6 above)
Comment (xuzhg) [reply src/Microsoft.OData.Edm/ExtensionMethods/ToTraceStringExtensionMethods.cs:140]: No, I should remove this line and change the codes to use the string Interpolation.  #Resolved

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 4 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:197]: see: https://github.com/OData/odata.net/blob/dev-8.x/src/Microsoft.OData.Core/Json/IJsonWriter.cs#L26, etc.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/SemanticAst/ODataPathExtensionsTests.cs (hunk 7 above)
Comment (xuzhg) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/SemanticAst/ODataPathExtensionsTests.cs]: Changed.

**Review by mikepizzo — APPROVED** [review]

**Pull request merged**